tests/test_umbral.py

```diff
@@ -18,15 +19,43 @@ def test_decapsulation_by_alice():
     priv_key = pre.gen_priv()
     pub_key = pre.priv2pub(priv_key)
 
-    sym_key, capsule = pre.encapsulate(pub_key)
+    sym_key, capsule = pre._encapsulate(pub_key)
     assert len(sym_key) == 32
 
     # The symmetric key sym_key is perhaps used for block cipher here in a real-world scenario.
 
-    sym_key_2 = pre.decapsulate_original(priv_key, capsule)
+    sym_key_2 = pre._decapsulate_original(priv_key, capsule)
     assert sym_key_2 == sym_key
 
 
+@pytest.mark.parametrize("N,threshold", parameters)
+def test_simple_api(N, threshold):
+    params = umbral.UmbralParameters()
+    pre = umbral.PRE(params)
+
+    priv_key_alice = keys.UmbralPrivateKey.gen_key(params)
+    pub_key_alice = priv_key_alice.get_pub_key(params)
+
+    priv_key_bob = keys.UmbralPrivateKey.gen_key(params)
+    pub_key_bob = priv_key_bob.get_pub_key(params)
+
+    plain_data = b'attack at dawn'
+    enc_data, capsule = pre.encrypt(pub_key_alice, plain_data)
+
+    dec_data = pre.decrypt(priv_key_alice, capsule, enc_data)
+    assert dec_data == plain_data
+
+    rekeys, _ = pre.split_rekey(priv_key_alice, pub_key_bob, threshold, N)
+    for rekey in rekeys:
+        cFrag = pre.reencrypt(rekey, capsule)
+        capsule.attach_cfrag(cFrag)
```

Comment:
Ah interesting. So we attach cFrag to exactly the original capsule, not a brand new one?

Reply:
You attach it, then call `capsule.reconstruct` which returns a new capsule.